beprof/curve.py

```
@@ -75,8 +78,21 @@ def y(self):
     def y(self, value):
         self[:, 1] = value
 
-    def rescale(self, factor=1.0):
-        self.y /= factor
+    def rescale(self, factor=1.0, allow_cast=True):
+        """
+        todo: write docstring nad doctests
+        :param factor:
+        :param allow_cast:
+        :return:
+        """
+        try:
+            self.y /= factor
+        except TypeError as e:
+            logger.error("Division in place is impossible.\n%s" % e)
+            if allow_cast:
+                self.y = self.y / factor
+            else:
+                logger.error("allow_cast flag set to True should help")
```

Does code execution continue or stops with some exception being rethrown ?

---

It stops, we could go for re-raise at the end - after calculating with ` self.y = self.y / factor` or ignore TypeErrors thrown by  `self.y /= factor` or... just log it?
```py
        try:
            self.y /= factor
        except TypeError as e:
            logger.error("Division in place is impossible")
            if allow_cast:
                self.y = self.y / factor
            else:
                logger.error("allow_cast flag set to True should help")
            raise e
```

---

Ignore error case (clean, but...):
```
>>> a
Curve([[1, 5],
       [2, 6],
       [3, 7],
       [4, 8]])
>>> a.rescale(2)
Division in place is impossible
>>> a
Curve([[1, 2],
       [2, 3],
       [3, 3],
       [4, 4]])
>>> a.rescale(2, allow_cast=False)
Division in place is impossible
allow_cast flag set to True should help
>>> a
Curve([[1, 2],
       [2, 3],
       [3, 3],
       [4, 4]])
```
Works like this for code below:
```py
    def rescale(self, factor=1.0, allow_cast=True):
        try:
            self.y /= factor
        except TypeError as e:
            logger.error("Division in place is impossible")
            if allow_cast:
                self.y = self.y / factor
            else:
                logger.error("allow_cast flag set to True should help")
```

---

The thing is - at least for me if something throws exception - it probably did not finish it's job (and possibly rolled back). So raising exception after successful division is... could be misinterpreted?